### PR TITLE
claims: Filter ipamclaim lookup according namespace

### PR DESCRIPTION
**Pull request by oshoval**
**What this PR does / why we need it**:
Currently looking for matching ipam claims is done only
according the owning VM name.
It means that when removing finalizer, we will remove
the finalizer from ipam claims with the same owning VM name across
all namespaces, which is wrong.
Fix it by limiting the search to the specific VM namespace.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by oshoval** [comment]
Alternative is to add the namespace to the list options

File: pkg/vminetworkscontroller/vmi_controller_test.go [hunk 1]
```diff
@@ -195,7 +195,7 @@ var _ = Describe("vmi IPAM controller", Serial, func() {
 					ObjectMeta: metav1.ObjectMeta{
 						Name:      "vm1.randomnet",
 						Namespace: "ns1",
-						Labels:    ownedByVMLabel(vmName),
+						Labels:    ownedByVMLabel("ns1", vmName),
```
Comment (maiqueb) [inline pkg/vminetworkscontroller/vmi_controller_test.go:198]: can't we use the constant here ?

**Review by maiqueb — COMMENTED** [review]
The changes does look good to me, good catch.

I wonder how can we add e2e tests for this entire finalizer effort. We should sync tomorrow about that.

**Comment by oshoval** [comment]
> I wonder how can we add e2e tests for this entire finalizer effort. We should sync tomorrow about that.

I am continuing the finalizer effort on https://github.com/kubevirt/ipam-extensions/pull/40
there are known issues with finalizers, so i believe it should be handle once it is fixed one way or another

File: pkg/vminetworkscontroller/vmi_controller_test.go (hunk 1 above)
Comment (oshoval) [reply pkg/vminetworkscontroller/vmi_controller_test.go:198]: since 2 lines above it is hardcoded i followed the same semantics

**Comment by oshoval** [comment]
> Alternative is to add the namespace to the list options but when having it part of the label, it adds small info to whoever looks on the label manually fwiw

@maiqueb 
wdyt ?

**Comment by oshoval** [comment]
failure here, rerunning, lets see if due to this PR or not (will check locally according needs as well)
https://github.com/kubevirt/ipam-extensions/actions/runs/9852423425/job/27200946613?pr=41

**Comment by oshoval** [comment]
used list filterting, lets see if it works this way, it is also simpler

**Comment by oshoval** [comment]
All passed now,
ptal

**Review by maiqueb — APPROVED** [review]
Makes sense to me. 

Would it / does it make sense to add a failing test first ?

**Comment by oshoval** [comment]
> Makes sense to me.
> 
> Would it / does it make sense to add a failing test first ?

I think it is too corner case that doesnt worth it,
i.e if the code had that in first place one wouldnt consider adding this specific cross talk test

**Comment by oshoval** [comment]
btw when you are fine with the PR, please also /approve
as tide requires it
Thanks
